### PR TITLE
[RW-8015][risk=no] new e2e Controlled Tier workspace test

### PR DESCRIPTION
**Pull request by aweng98**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/app/component/card/workspace-card.ts [hunk 1]
```diff
@@ -103,6 +103,16 @@ export default class WorkspaceCard extends BaseCard {
     return wholeText.replace('Last Changed: ', '').trim();
   }
 
+  async controlledTiersIconExists(): Promise<boolean> {
```
Comment (jmthibault79) [inline e2e/app/component/card/workspace-card.ts:106]: Tiers -> Tier

File: e2e/tests/controlled-tier/ct-workspace-share.spec.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import UnavailableTierModal from 'app/modal/unavailable-tier-modal';
+import DataAccessRequirementsPage from 'app/page/data-access-requirements-page';
+
+describe('Share Controlled Tier Workspace', () => {
+  const hasCTAccessUser = config.WRITER_USER;
+  const notCTUser = config.EGRESS_TEST_USER; // Without CT access
+
+  const workspaceName = makeRandomName('e2eShareControlledTierWorkspace');
+
+  test('User without CT access cannot create CT workspace', async () => {
```
Comment (jmthibault79) [inline e2e/tests/controlled-tier/ct-workspace-share.spec.ts:20]: This is a nice test, thanks!  I would also like to see the flow for cancel modal - observe that the Create Workspace page forces the tier back to registered.

Don't need a new test for this - we can add this as an earlier step here.

File: e2e/tests/controlled-tier/ct-workspace-share.spec.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    const aboutPage = new WorkspaceAboutPage(page);
+    await openTab(page, Tabs.About, aboutPage);
+
+    await aboutPage.selectWorkspaceAction(MenuOption.Share, { waitForNav: false });
+    let shareModal = new ShareModal(page);
+    await shareModal.waitForLoad();
+
+    // Search for user without CT access returns empty result in Share modal.
+    const userFound = await shareModal.userExists(notCTUser);
+    expect(userFound).toBe(false);
```
Comment (jmthibault79) [inline e2e/tests/controlled-tier/ct-workspace-share.spec.ts:61]: nice

File: e2e/tests/controlled-tier/ct-workspace-share.spec.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    await shareModal.clickButton(LinkText.Cancel, { waitForClose: true });
+    await waitWhileLoading(page);
+
+    // Share with a WRITER
+    await aboutPage.selectWorkspaceAction(MenuOption.Share, { waitForNav: false });
+    shareModal = new ShareModal(page);
+    await shareModal.shareWithUser(hasCTAccessUser, WorkspaceAccessLevel.Writer);
+  });
+
+  test('Writer can open and duplicate CT workspace', async () => {
```
Comment (jmthibault79) [inline e2e/tests/controlled-tier/ct-workspace-share.spec.ts:72]: I would like to reword this slightly.  I'm reading this as a test which actually DOES these actions.  But this test actually shows that the user has the correct access to do these actions.

Comment (jmthibault79) [reply e2e/tests/controlled-tier/ct-workspace-share.spec.ts:96]: cool

**Review by jmthibault79 — APPROVED** [review]

**Review by dmohs — COMMENTED** [review]
Feedback for us to discuss if you'd like, but no need to create a follow-up PR.

**Pull request merged**